src/Behat/Config/Config.php

```diff
@@ -38,8 +51,47 @@ public function withProfile(Profile $profile): self
         return $this;
     }
 
+    public function withPreferredProfile(string $profileName): self
+    {
+        $this->settings[self::PREFERRED_PROFILE_NAME_SETTING] = $profileName;
+
+        return $this;
+    }
+
     public function toArray(): array
     {
         return $this->settings;
     }
+
+    public function toPhpExpr(): Node\Expr
+    {
+        $configObject =  $this->builderFactory->new(new FullyQualified(self::class));
+        $expr = $configObject;
+
+        foreach ($this->settings as $settingsName => $settings) {
+            if ($settingsName === self::PREFERRED_PROFILE_NAME_SETTING) {
+                $args = $this->builderFactory->args([$settings]);
+                $expr = $this->builderFactory->methodCall($expr, self::PREFERRED_PROFILE_FUNCTION, $args);
+            } elseif ($settingsName === self::IMPORTS_SETTING) {
+                if (count($settings) === 1) {
+                    $args = $this->builderFactory->args([$settings[0]]);
+                } else {
+                    $args = $this->builderFactory->args([$settings]);
+                }
+                $expr = $this->builderFactory->methodCall($expr, self::IMPORT_FUNCTION, $args);
+            } else {
+                $profile = new Profile($settingsName, $settings ?? []);
```

Comment:
In many cases we will build a new Config object (in this case a Profile object) and call its toPhpExpr to build an expression which can be used on a function call on the current object (in this case a withProfile() call